### PR TITLE
Simplify PyGMT Release process

### LINKED ISSUE #444: Add a PyGMT release checklist
**Description of the desired feature**

Like [what we did in the GMT repository](https://github.com/GenericMappingTools/gmt/blob/master/.github/ISSUE_TEMPLATE/release_checklist.md), we could have a release checklist  for PyGMT, which can simplify/standardize the process of new releases.

The checklist template should be placed in the `.github/ISSUE_TEMPLATE` directory.

### PR DESCRIPTION
**Pull request by weiji14**
**Description of proposed changes**

Releases can take a lot of time, so best to streamline the process so that we don't get put off by it!

This PR does two things:
- [ ] Adds a 'Release' checklist as an issue template, similar to [upstream GMT's](https://github.com/GenericMappingTools/gmt/blob/master/.github/ISSUE_TEMPLATE/release_checklist.md.
)
- [ ] Automate Changelog writing (somewhat) using a [Release Drafter bot](https://github.com/apps/release-drafter).

In the future, we'll need to label our Pull Requests accordingly (e.g. as bug, enhancement, documentation, etc), and this should keep a running draft changelog under https://github.com/GenericMappingTools/pygmt/releases.

Fixes #444, #443.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/release_checklist.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+**Scheduled Date**: YYYY/MM/DD
+
+**Before release**:
+- [ ] Reserve a DOI on [Zenodo](https://zenodo.org)
+- [ ] Update Changelog
+
+**Release**:
+- [ ] Make a tag and push it to Github
+```
+git tag x.x.x
```
Comment (seisman) [inline .github/ISSUE_TEMPLATE/release_checklist.md:20]:
```suggestion
git tag vx.x.x
```

File: .github/workflows/release-drafter.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+name: Release Drafter
+
+on:
+  push:
+    # branches to consider in the event; optional, defaults to all
+    branches:
+      - master
```
[the remark this answers is not in the document]
Comment (weiji14) [reply .github/workflows/release-drafter.yml:7]: Maybe not now, since we're not using `x.x` branches yet.

[the remark this answers is not in the document]
Comment (seisman) [reply .github/workflows/release-drafter.yml:7]: OK.

File: .github/workflows/release-drafter.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  update_release_draft:
+    runs-on: ubuntu-latest
+    steps:
+      # Drafts your next Release notes as Pull Requests are merged into "master"
+      - uses: release-drafter/release-drafter@v5.8.0
+        with:
+          # (Optional) specify config name to use, relative to .github/. Default: release-drafter.yml
+          # config-name: my-config.yml
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (weiji14) [inline .github/workflows/release-drafter.yml:19]: Not quite sure if we need to set a secret for this? I'll be in a meeting soon, will check back after.

Comment (weiji14) [reply .github/workflows/release-drafter.yml:19]: Seems to be set automatically for every Github Action, see https://help.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token. All good then.

**Comment by weiji14** [comment]
Ok, need to remove the `with` since it's not used...

**Comment by weiji14** [comment]
Awesome, it's working over at https://github.com/GenericMappingTools/pygmt/releases!

Release Drafter Github Action drafting release notes

**Pull request merged**